sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveAliasesSuite.scala

```diff
@@ -88,4 +94,46 @@ class ResolveAliasesSuite extends AnalysisTest {
     checkAliasName(t1.select(DateSub(Literal(Date.valueOf("2021-01-18")), Literal(null))),
       "date_sub(DATE '2021-01-18', NULL)")
   }
+
+  test("SPARK-40822: Stable derived column aliases") {
+    withSQLConf(SQLConf.STABLE_DERIVED_COLUMN_ALIAS_ENABLED.key -> "true") {
+      Seq(
+        // Literals
+        "' 1'" -> "' 1'",
+        """"abc"""" -> """"abc"""",
+        """'\t\n xyz \t\r'""" -> """'\t\n xyz \t\r'""",
+        "1l" -> "1L", "1S" -> "1S",
+        "date'-0001-1-28'" -> "date'-0001-1-28'",
+        "interval 3 year 1 month" -> "INTERVAL3YEAR1MONTH",
+        "x'00'" -> "x'00'",
+        // Preserve case
+        "CAST(1 as tinyint)" -> "CAST(1AStinyint)",
```

Comment:
ditto, all types should be keyword.

Reply:
This PR https://github.com/apache/spark/pull/40565 does this.